### DISCUSSION TITLE
DicomDirectory - going from fo-dicom v3 to v4

### DISCUSSION THREAD

**Discussion by lukegabello** [fo-dicom/fo-dicom#1165] [Q&A]
I am upgrading to fo-dicom v4, and I noted a couple of changes in DicomDirectory that I could use some advice on. The code I am working with used DicomDirectory's Find and Remove methods that were available in v3. Does anyone have a suggestion on what to substitute these methods with? I could engineer the two methods myself, but I prefer to stick to using the framework as much as possible. Thank you ahead of time!

**Comment by gofal** [comment]
I don't know version 3 interface very well. What did those methods do exactly?
Here is an example of how to use DicomDirectory in version 4: https://github.com/fo-dicom/fo-dicom-samples/blob/master/Desktop/DICOM%20Media/Program.cs
